tests/unit/s2n_ktls_mode_test.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_ktls.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* Default config kTLS mode */
+    {
+        DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+        EXPECT_NOT_NULL(config);
+        EXPECT_FALSE(config->ktls_send_requested);
+        EXPECT_FALSE(config->ktls_recv_requested);
+    };
+
+    /* Request config kTLS mode
+     *
+     * Requesting modes is additive and Duplex means both Send and Recv are requested
+     */
+    {
+        DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+        EXPECT_NOT_NULL(config);
+
+        EXPECT_SUCCESS(s2n_config_set_ktls_mode(config, S2N_KTLS_MODE_SEND));
+        EXPECT_TRUE(config->ktls_send_requested);
+        EXPECT_FALSE(config->ktls_recv_requested);
+
+        EXPECT_SUCCESS(s2n_config_set_ktls_mode(config, S2N_KTLS_MODE_RECV));
+        EXPECT_TRUE(config->ktls_send_requested);
```

Comment:
It seems odd to me to that requesting `S2N_KTLS_MODE_RECV` doesn't disable `S2N_KTLS_MODE_SEND`. Is there a reason to have `S2N_KTLS_MODE_DUPLEX` with this behavior? If you wanted to switch from send to recv (not sure if this is even a use case), you would have to first request `S2N_KTLS_MODE_DISABLED` and then `S2N_KTLS_MODE_RECV`, which seems unintuitive. Maybe this is intended/not an issue, though.

Reply:
Yea fair point. I'll change this behavior!

Reply:
updated https://github.com/aws/s2n-tls/pull/3797/commits/f0170b4f6adb44f24513c0c121a023f548138ecc